clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -498,6 +517,44 @@ public void send(String method, Object... args) {
         return subject;
     }
 
+    public <T> Observable<T> stream(Class<T> returnType, String target, Object ... args) {
+        String invocationId = connectionState.getNextInvocationId();
+        StreamInvocationMessage streamInvocationMessage = new StreamInvocationMessage(invocationId, target, args);
+        InvocationRequest irq = new InvocationRequest(returnType, invocationId);
+        connectionState.addInvocation(irq);
+        ReplaySubject<T> subject = ReplaySubject.create();
+
+        Action1<HubMessage> streamInvocationCallbackHandler = (hubMessage) -> {
+            switch (hubMessage.getMessageType()) {
+                case STREAM_ITEM:
+                    StreamItem streamItem = (StreamItem)hubMessage;
+                    subject.onNext((T)streamItem.getResult());
+                    break;
+                case COMPLETION:
+                    CompletionMessage completionMessage = (CompletionMessage)hubMessage;
+                    if (completionMessage.getError() != null) {
+                        subject.onError(new Exception(completionMessage.getError()));
+                    }
+                    else {
+                        subject.onComplete();
+                    }
+                    break;
+                default:
+                    throw new RuntimeException("Unexpected Message type");
+            }
+        };
+
+        ActionBase action = params -> streamInvocationCallbackHandler.invoke(HubMessage.class.cast(params[0]));
+        handlers.put(invocationId, action, returnType);
```

Comment:
We should refactor this code to use `irq.getPendingCall()` instead of `handlers.put`